### ISSUE TITLE
What is the point of signing key pairs?

### ISSUE THREAD

**Issue by dsferruzza** [IronCoreLabs/recrypt-rs#176]
Hi!

I think I understand how to use the transform encryption feature of this lib, thanks to the documentation example.
But I do not understand why we need a signing key pair when encrypting or generating a transform key.
According to my tests, encryption/transformation/decryption seems to work even if I generate a new signing key pair every time one is needed.

So I wonder:

- Is it OK to do that? (Generate new key pairs and not keeping them)
- Could you explain/document what is the point of these keys and how to use them?

Have a nice day!

**Comment by BobWall23** [comment]
TLDR: yes, it is OK to do that. Below is an explanation of how we use those keys.

Each encrypted data block or transform key is signed to ensure that it is not corrupted. The public key for the signing key pair is included with the signed value, so it would be possible to generate an ephemeral key pair for each encryption or transform key generation. If you do that, you lose the ability to identify the user that did the encryption or generated the transform key (unless you track that association separately, in which case there is no benefit to not using a single key pair associated with that user). It also takes time to generate the key pair, and in our use cases, we have a signing key pair already, so we just use it for each encryption and transform key generation. If you generate a new key pair for each encryption, it will work correctly.

**Comment by BobWall23** [comment]
I'm going to close the issue - please reopen if you have further questions.

**Comment by dsferruzza** [comment]
This is much clearer! Thanks a lot!
